### PR TITLE
webhook/configmap: mutate BinaryData as well

### PR DESCRIPTION
**Pull request by rrondeau**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| API breaks?     | no
| Deprecations?   | no
| Related tickets | 
| License         | Apache 2.0

### What's in this PR?

This PR adds the possibility to mutate the binary data of configmaps.
Binary data must be base64 encoded inside vault (https://github.com/hashicorp/vault/issues/1423)

### Why?

We need to mutate configMaps with binary data

### Additional context

This only work with base64 encoded data inside vault since vault cannot store binary data without encoding

### CODE REVIEW

**Comment by rrondeau** [comment]
Hi
I put an example into the deploy directory like you ask.
If you want some other docs / example, i will be happy to oblige.

**Comment by bonifaido** [comment]
Thanks @rrondeau , we are having some Circle CI seats issue right now, under investigation, thanks!

File: deploy/test-configmap.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+apiVersion: v1
+kind: configMap
```
Comment (pepov) [inline deploy/test-configmap.yaml]:
```suggestion
kind: ConfigMap
```

File: cmd/vault-secrets-webhook/configmap.go [hunk 2]
```diff
@@ -28,6 +29,11 @@ func configMapNeedsMutation(configMap *corev1.ConfigMap) bool {
 			return true
 		}
 	}
+	for _, value := range configMap.BinaryData {
+		if strings.HasPrefix(string(value), "vault:") {
```
Comment (pepov) [inline cmd/vault-secrets-webhook/configmap.go]: binaryData is automatically base64 decoded here?

File: deploy/test-configmap.yaml (hunk 1 above)
Comment (rrondeau) [reply deploy/test-configmap.yaml]: Yep sorry for the typo

File: cmd/vault-secrets-webhook/configmap.go (hunk 2 above)
Comment (rrondeau) [reply cmd/vault-secrets-webhook/configmap.go]: yes like secrets data

**Comment by bonifaido** [comment]
CircleCI has deployed a fix to resolve our issues, thanks for your patience, and for the contribution of course!

**Pull request merged**